src/main.rs

```diff
@@ -96,15 +96,26 @@ fn spawn_cargo(
     cmd.arg("--");
 
     // Rustc flags.
-    // We care about asm.
-    cmd.args(["--emit", syntax.emit()])
+    cmd
+        // Start with the user-supplied codegen flags, which we might need to override.
+        .args(cargo.codegen.iter().flat_map(|c| ["-C", c]))
+        // Next, we care about asm/wasm/llvm-ir/llvm-mac.
+        .args(["--emit", syntax.emit()])
         // So only one file gets created.
         .arg("-Ccodegen-units=1")
-        // Debug info is needed to map to rust source.
-        .arg("-Cdebuginfo=2")
         .args(syntax.format().iter().flat_map(|s| ["-C", s]))
         .args(target_cpu.iter().map(|cpu| format!("-Ctarget-cpu={cpu}")));
 
+    // Debug info is needed to map asm/wasm output to rust source.
+    if format.rust
```

Comment:
I don't, so I'll split that change into a separate PR.

When does that function boundary detection take place? I can try to look into, but it's not obvious to me when that might happen for `!format.rust`.

Reply:
This is happening around `Label` parsing here: `src/asm/statements.rs`. Looks like it broke Windows too.

Reply:
So it's an ASM-only thing? 

I originally ignored whether `--rust` had been set... maybe I should go back to doing that? In some cases (WASM and ASM on Unix) `debuginfo=2` could still be set unnecessarily, but that's probably ok.

Reply:
Code is not doing anything super fancy when deciding where functions are - it looks for some lines which I got by looking at asm myself and adjusted a few times to handle Windows/MacOS. Not having debug info means those lines are slightly different, #137 is caused by something similar/same. A proper fix would be to figure out the difference and support both with/without debug into on all major platforms. Problem with that is that this is a bunch of work, more so because I only have a Linux machine.

Stuff that currently breaks is asm specific, yes.

Reply:
I too am mostly only using Linux.

But good news: trivial cross compilation seems to work because no linking actually takes place when outputting ASM. Or at least when looking into the failing example... I'm not sure if in other cases it might still be necessary for some reason.

Bad news: on Windows it doesn't look like it would be very easy to detect function boundaries without at least `debuginfo=1`:

```asm
; debuginfo=0
_ZN9shadow_rs3add17h8880170e08047822E:
        lea     rax, [rcx + rdx]
        ret
```

```asm
; debuginfo=1 or debuginfo=2
_ZN9shadow_rs3add17h8880170e08047822E:
.Lfunc_begin0:
        .cv_func_id 0
        .cv_file        1 "/home/jonas/Code/cargo-show-asm/sample_rlib/lib.rs" "7A28BF0F943AD15B0D3987FC26281016667F7A8A" 2
        .cv_loc 0 1 2 0
        lea     rax, [rcx + rdx]
        .cv_loc 0 1 3 0
        ret
.Ltmp0:
.Lfunc_end0:
```

---

In any case, have you considered my suggestion of keeping `debuginfo=2` for ASM and, for simplicity, WASM output, regardless of `--rust`?

In the meantime, I'll split the debuginfo commit into a separate commit.

Reply:
> In any case, have you considered my suggestion of keeping debuginfo=2 for ASM and, for simplicity, WASM output, regardless of --rust?

Yes, this sounds good.